src/v_video.c

```diff
@@ -849,9 +871,39 @@ void V_ScreenShot(char *format)
 #ifdef HAVE_LIBPNG
     if (png_screenshots)
     {
-    WritePNGfile(lbmname, I_VideoBuffer,
-                 SCREENWIDTH, SCREENHEIGHT,
-                 W_CacheLumpName (DEH_String("PLAYPAL"), PU_CACHE));
+        byte * buf, * oldbuf;
+        int oldpitch;
+
+        // Prepare a temporary buffer to write the screenshot to
+        buf = malloc(screenshot_width * screenshot_height);
+
+        if(!buf)
+        {
+            return;
+        }
+
+        // Ideally we'd do this only once, but V_Init is called too early
+        // and screenshot_width/height haven't been read from the config
+        V_InitScreenShotScaler();
+
+        // back up the scaling system's current output buffer settings
+        oldbuf = I_GetScaleDestBuf();
```

Comment:
Yeah, it's not about preserving vanilla code at all - you seemed to have missed my point. I'm more concerned with writing code that's nice and that follows a clear abstraction boundary. Trying to avoid complexity is actually my goal here - having Get/Set methods that allow external modules to poke into the internals of the i_scale.c code is not nice, in this respect.